tests/phpunit/CRM/Member/Form/MembershipRenewalTest.php

```diff
@@ -700,4 +700,50 @@ protected function getBaseSubmitParams() {
     ];
   }
 
+  /**
+   * Test renewing an expired membership.
```

Comment:
@eileenmcnaughton Can we get a description of what should *actually* happen. I realise you can more or less infer from the code in the test but much easier if we have in the description.  I've come across this recently on a real site so am aware that Expired->Current and dates go to now/now + 1 term I think?

Reply:
Good goal - when I know I'll let you know - I'm locking in the results I get back in the test but I haven't analysed them much, mostly because the goal is to clean stuff up & know it's not changing anything

Reply:
Ok, but I'm not sure how we can merge a test if we don't know what it is testing? Looking through the code it is not clear to me what the new dates on the membership actually are. My experience (renewing memberships via the frontend) is that the membership start date will match the contribution receive date + 1 day and the end date will be +1 term. It seems that this form is behaving very differently.

Reply:
@mattwire right - the point of the test is to establish that existing behaviour is tested. Before we change code - even code we understand poorly - we add a test. Then we clean up the code, ensuring that the outcome doesn't change. It might be that we decide to change behaviour - the we change the test.

I don't have a clue why this form does things the way it does - or what most of the code is doing. So the requirement is that before I rationalise this chunk of code I ensure I don't inadvertantly change the behaviour. This is our standard TDD code cleanup process